Functions/FirebaseFunctions/FUNContext.m

```diff
@@ -101,7 +101,10 @@ - (void)getContext:(void (^)(FUNContext *context, NSError *_Nullable error))comp
 
     [_appCheck getTokenForcingRefresh:NO
                            completion:^(id<FIRAppCheckTokenResultInterop> _Nonnull tokenResult) {
-                             appCheckToken = tokenResult.token;
+                             // Send only valid token to functions.
+                             if (tokenResult.error == nil) {
+                               appCheckToken = tokenResult.token;
```

Comment:
Why does the `getTokenForcingRefresh` return a token when there are errors?

A Swift Result type could help this API.  😛

Reply:
The current App Check semantic doesn't exactly fit Swift Result type. For most of the integrator SDKs setting a value to FAC header even in the case of the failure is a way to indicate that App Check is integrated with the app (see more details in go/app-attestation-sdk-api#heading=h.605kjfzcbxum). To simplify integration for these products it was decided to always return a token even if an error happened (it is a dummy token in the case of the error). 

For Functions the dummy token is useless and it appeared to trigger another issue because of the way Firebase Functions wrapper handle invalid tokens. So we are fixing it by:
1. Not sending the dummy tokens for Functions on the SDK side
2. Fixing handling of invalid FAC tokens on the callable functions side